src/adapters/org.mozilla-iot.adapters.gpio/index.js

```diff
@@ -10,7 +10,7 @@
 
 var fs = require('fs');
 
-function maybeLoadGpioAdapter(adapterManager, adapterId, adapterConfig) {
+function maybeLoadGpioAdapter(adapterManager, manifest) {
```

Comment:
We probably need a mechanism for the adapter to leave a message that indicates that it couldn't be loaded and why not.

For example, the gpio adapter will fail to load on Mac OSX and on most linux laptops, since they don't actually have a gpio subsystem.

If I enable gpio on my linux laptop, then all that happens is that there is a console message printed:
```
Not starting GPIO adapter - no permissions to /sys/class/gpio/export
```